akka-stream-tests/src/test/scala/akka/stream/scaladsl/SynchronousFileSourceSpec.scala

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import java.io.{ File, FileWriter }
+import java.util.Random
+
+import akka.stream.testkit.{ AkkaSpec, StreamTestKit }
+import akka.stream.{ ActorFlowMaterializer, ActorFlowMaterializerSettings }
+import akka.util.ByteString
+
+import scala.concurrent.Await
+
+class SynchronousFileSourceSpec extends AkkaSpec(
+  """ akka.actor.default-mailbox.mailbox-type = "akka.dispatch.UnboundedMailbox" """) {
+
+  val settings = ActorFlowMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorFlowMaterializer(settings)
+
+  val testFile = {
+    val f = File.createTempFile("file-source-spec", "tmp")
+    new FileWriter(f)
+      .append("a" * 10)
+      .append("b" * 10)
+      .append("c" * 10)
+      .append("d" * 10)
+      .append("e" * 10)
+      .append("f" * 10)
+      .close()
+    f
+  }
+
+  val LinesCount = 2000 + new Random().nextInt(300)
+
+  val manyLines = {
+    val f = File.createTempFile(s"file-source-spec-lines_$LinesCount", "tmp")
+    val w = new FileWriter(f)
+    (1 to LinesCount).foreach { l ⇒
+      w.append("a" * l).append("\n")
+    }
+    w.close()
+    f
+  }
+
+  override def afterTermination(): Unit = {
+    testFile.delete()
+    manyLines.delete()
+  }
+
+  "File Source" must {
+    "read lines with readAhead 2" in {
+      val chunkSize = 10
+      val readAhead = 2
+
+      val p = Source.synchronousFile(testFile, chunkSize).runWith(Sink.publisher)
+      val c = StreamTestKit.SubscriberProbe[ByteString]()
+      p.subscribe(c)
+      val sub = c.expectSubscription()
+
+      sub.request(3)
+      c.expectNext().utf8String should ===("a" * 10)
+      c.expectNext().utf8String should ===("b" * 10)
+      c.expectNext().utf8String should ===("c" * 10)
+      sub.request(1)
+      c.expectNext().utf8String should ===("d" * 10)
+      sub.request(1)
+      c.expectNext().utf8String should ===("e" * 10)
+      sub.request(100)
+      c.expectNext().utf8String should ===("f" * 10)
+
+      c.expectComplete()
+    }
+
+    List(
+      Settings(chunkSize = 8, readAhead = 8),
+      Settings(chunkSize = 128, readAhead = 1),
+      Settings(chunkSize = 256, readAhead = 1),
+      Settings(chunkSize = 256, readAhead = 2),
+      Settings(chunkSize = 512, readAhead = 1),
+      Settings(chunkSize = 512, readAhead = 2),
+      Settings(chunkSize = 4096, readAhead = 1)) foreach { settings ⇒
```

Comment:
this test case was more interesting for the async impl, I'll lessen the test cases we do for the synchronous one here.